tensorflow_serving/batching/batching_util.cc

```diff
@@ -0,0 +1,229 @@
+// is ignored, because we don't perform padding in that dimension.
+template<int num_dims>
+std::array<OneDimPadding, num_dims> CreatePadding(Tensor tensor,
+        const std::vector<int>& max_dim_sizes) {
+  std::array<OneDimPadding, num_dims> padding;
+  for (unsigned int i = 0; i < max_dim_sizes.size(); ++i) {
+    if (i > 0 && max_dim_sizes[i] - tensor.dim_size(i) > 0) {
+      padding[i] = {0, max_dim_sizes[i] - tensor.dim_size(i)};
+    } else {
+      padding[i] = {0, 0};
+    }
+  }
+  return padding;
+}
+
+// Functor, which performs padding of given input tensor
+// using specified padding signature.
+// For example, given tensor of shape [1, 2, 3] and padding signature
+// [[0, 0], [0, 2], [2, 2]]
+// functor produces padded_tensor of shape [1, 4, 7].
+template <typename T, int num_dims>
+struct PadTensor {
+  Status operator()(Tensor input,
+      const std::array<OneDimPadding, num_dims>& padding,
+      Tensor* output) {
+    TensorShape output_shape;
+    for (int d = 0; d < num_dims; ++d) {
+      // Pad before existing elements.
+      const int32 before_d = padding[d].first;
+      // Pad after existing elements.
+      const int32 after_d = padding[d].second;
+      output_shape.AddDim(before_d + input.dim_size(d) + after_d);
+    }
+    if (output_shape.num_elements() == input.NumElements()) {
+      auto result = output->CopyFrom(input, output_shape);
+      if (!result) {
+        return errors::Internal("Couldn't create output.");
+      }
+      return Status::OK();
+    }
+    *output = Tensor(input.dtype(), output_shape);
+    typename TTypes<T, num_dims>::Tensor inputs = input.tensor<T, num_dims>();
+    T pad_value(input.flat<T>()(0));  // using existing values in padding
+    output->tensor<T, num_dims>() = inputs.pad(padding, pad_value);
+    return Status::OK();
+  }
+};
+
+// functor specialization for scalars:
+// we do not perform padding in that case.
+template<typename T>
+struct PadTensor<T, 0> {
+  Status operator()(Tensor input,
+      const std::array<OneDimPadding, 0>&  padding, Tensor* output) {
+    TensorShape output_shape;
+    *output = Tensor(input.dtype(), output_shape);
+    typename TTypes<T, 0>::Tensor inputs = input.tensor<T, 0>();
+    output->tensor<T, 0>() = inputs;
+    return Status::OK();
+  }
+};
+
+// Invokes padding procedure for specific tensor ranks.
+// Only ranks from 0 to 6 are supported (like in PadOp).
+template<typename T>
+Status PadTensorOfSpecificType(const Tensor& tensor,
+                               const std::vector<int>& max_dim_sizes,
+                               Tensor* output_tensor) {
+  int num_dims = tensor.dims();
+  Status padding_status;
+  switch (num_dims) {
+    case 0: {
+      std::array<OneDimPadding, 0> padding;
+      padding = CreatePadding<0>(tensor, max_dim_sizes);
+      PadTensor<T, 0> padding_functor = PadTensor<T, 0>();
+      padding_status = padding_functor(tensor, padding, output_tensor);
```

Comment:
You can eliminate the padding_status variable and just in-line the returns, so:
return padding_functor(...);
instead of
padding_status = padding_functor(...);
break;
... and returning after the switch statement.